polar/src/formatting.rs

```diff
@@ -299,7 +299,11 @@ pub mod to_polar {
 
     impl ToPolarString for ExternalInstance {
         fn to_polar(&self) -> String {
-            format!("^{{id: {}}}", self.instance_id)
+            if let Some(ref repr) = self.repr {
+                repr.clone()
+            } else {
+                format!("^{{id: {}}}", self.instance_id)
```

Comment:
Hmm, actually not sure this should go in ToPolarString... It's not quite correct. ToPolarString should be parseable back into a term.

Reply:
`ExternalInstance` was and is the exception to that. There is no parseable notation for an external instance.

Reply:
Oh you're totally right, I thought we had parser code still for `^{...}`. I think this is still unfortunate though. Other than implementing a _new_ format trait which uses ToPolarString for most things, I can't see a good way to solve this. We should revisit this though.

Reply:
I don't think it's a problem. E.g., in Python, the default instance repr is not parsable; Common Lisp has `print-unreadable-object` for the same reason. Most instances don't have a parsable representation in most languages.